src/storage_pool.cc

```diff
@@ -116,7 +117,16 @@ NAN_METHOD(StoragePool::LookupByUUID)
 NAN_METHOD(StoragePool::LookupByVolume)
 {
   Nan::HandleScope scope;
-  return;
+  auto hv = ObjectWrap::Unwrap<Hypervisor>(info.This())->handle_;
+  auto volume = ObjectWrap::Unwrap<StorageVolume>(info[0]->ToObject())->handle_;
+  
+  Worker::Queue(info[1], [=](Worker::SetOnFinishedHandler onFinished) {
```

Comment:
@mbroadst remember that callback is not always on a static position, consider all the cases where flags (or other paramters) are optional. What's the value of repetitive code to retrieve the callback if it's always at the last position? So if anything, it would be `ASSERT_CALLBACK(info[info.Length() -1])`. and then it would make more sense to do `ASSERT_CALLBACK_IS_LAST_ARG(info)`. Now however we'd like to get rid of a macro and have a function, so why not do the very same thing inside `Worker::QueueCallback`? :-)

btw. I was doing some thinking over night and it seems there is no reason not to bind `info.This()` as permanent every time. On my PC I just found that having this last parameter as optional makes it crash:
```
  static void Queue(v8::Local<v8::Value> v8_callback, ExecuteHandler handler,
      v8::Local<v8::Object> parent = v8::Local<v8::Object>());
```
If it was named `QueueCallback` and took https://github.com/nodejs/nan/blob/master/doc/methods.md#api_nan_function_callback_info as argument then it could bind `This` automatically.

And the risk of not binding `This()` is something being garbage collected .. so having it done automatically would be a value. Unfortunately the whole `Nan:::FunctionCallbackInfo` is a template so I don't know if it will work having it defined as a parameter.